Packet++/header/WireGuardLayer.h

```diff
@@ -0,0 +1,257 @@
+	 */
+#pragma pack(push, 1)
+	struct wg_common_header
+	{
+		/** Message type field */
+		uint8_t messageType;
+		/** Reserved field (3 bytes) */
+		uint8_t reserved[3];
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_handshake_initiation
+	 * Represents the Handshake Initiation message
+	 */
+#pragma pack(push, 1)
+	struct wg_handshake_initiation
+	{
+		wg_common_header common;               ///< Common header for all WireGuard messages
+		uint32_t senderIndex;                  ///< Sender index
+		uint8_t initiatorEphemeral[32];        ///< Initiator's ephemeral public key
+		uint8_t encryptedInitiatorStatic[48];  ///< Encrypted initiator's static key
+		uint8_t encryptedTimestamp[28];        ///< Encrypted timestamp
+		uint8_t mac1[16];                      ///< MAC1 field
+		uint8_t mac2[16];                      ///< MAC2 field
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_handshake_response
+	 * Represents the Handshake Response message
+	 */
+#pragma pack(push, 1)
+	struct wg_handshake_response
+	{
+		wg_common_header common;         ///< Common header for all WireGuard messages
+		uint32_t senderIndex;            ///< Sender index
+		uint32_t receiverIndex;          ///< Receiver index
+		uint8_t responderEphemeral[32];  ///< Responder's ephemeral public key
+		uint8_t encryptedEmpty[16];      ///< Encrypted empty field
+		uint8_t mac1[16];                ///< MAC1 field
+		uint8_t mac2[16];                ///< MAC2 field
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_cookie_reply
+	 * Represents the Cookie Reply message
+	 */
+#pragma pack(push, 1)
+	struct wg_cookie_reply
+	{
+		wg_common_header common;      ///< Common header for all WireGuard messages
+		uint32_t receiverIndex;       ///< Receiver index
+		uint8_t nonce[24];            ///< Nonce field
+		uint8_t encryptedCookie[32];  ///< Encrypted cookie
+	};
+#pragma pack(pop)
+
+	/**
+	 * @struct wg_transport_data
+	 * Represents the Transport Data message
+	 */
+#pragma pack(push, 1)
+	struct wg_transport_data
+	{
+		wg_common_header common;   ///< Common header for all WireGuard messages
+		uint32_t receiverIndex;    ///< Receiver index
+		uint64_t counter;          ///< Counter field
+		uint8_t encryptedData[0];  ///< Flexible array member for encrypted data
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class WireGuardLayer
+	 * Represents a WireGuard protocol layer
+	 */
+	class WireGuardLayer : public Layer
+	{
+	public:
+		/**
+		 * Constructs a WireGuardLayer object.
+		 *
+		 * @param data Pointer to the raw data representing the WireGuard layer
+		 * @param dataLen Length of the data
+		 * @param prevLayer Pointer to the previous layer in the packet (if any)
+		 * @param packet Pointer to the packet this layer belongs to
+		 */
+		WireGuardLayer(uint8_t* data, size_t dataLen, Layer* prevLayer, Packet* packet)
+		    : Layer(data, dataLen, prevLayer, packet, WIREGUARD)
+		{
+			m_Data = data;
+			m_DataLen = dataLen;
+		}
```

Comment:
Why is `m_data` and `m_DataLen` being set again? Passing the pointers to the `Layer` ctor already sets the fields in:
https://github.com/seladb/PcapPlusPlus/blob/2270bd73521e20340a57792a9eda16d63638b058/Packet%2B%2B/header/Layer.h#L214-L217

Reply:
Good catch, I’ll remove the redundant assignments.